File: BankAccountProject_In_CPlusPlus/SavingsAccount.cpp
/*
Name: Olivia Nye 
Date: 04/05/2019
Class: CMPS 1600 Spring 2019
Lab 5: First C++ program, BankAccount with inheritence and manual checking 

SavingsAccount.cpp  constructs SavingsAccount as an extension of BankAccount and defines SavingsAccount function addInterest
*/

#include "SavingsAccount.h"
#include <iostream>
#include <string>

using namespace std;

//SavingsAccount constructor + extension of BankAccount
SavingsAccount:: SavingsAccount(string n, double b, double i) : BankAccount(n, b){
  interestRate = i;

}

//function calculates the amount of interest earned for the balance of the SavingsAccount depending on the value of the interestRate specified in the constructor and adds the interest to the balance by depositing it with the inherited deposit function from super class BankAccount
void SavingsAccount::addInterest(){
  double interest;
  interest = getBalance() * interestRate;
  deposit(interest);
}

File: BankAccountProject_java/SavingsAccount.java
//Olivia Nye -- CMPS 1600 Lab 1/2? Bank Account
public class SavingsAccount extends BankAccount {
	

	
	protected double interestRate;
	
	public SavingsAccount(String n, double b, double i) {
		super(n, b);
		interestRate = i / 100.0;
	}
		
	
	public void addInterest(){
						//calculate interest				//add it to balance
		this.balance = (super.getBalance() * interestRate) + super.getBalance();
	
	}
	
}


